src/replication/replicator.rs

```diff
@@ -376,6 +376,12 @@ impl Replicator {
                 )?
                 .messages
             }
+            proto::MessageType::LendStorage => {
+                stores
```

Comment:
@adityapk00 do these messages go through the engine validations? I added a validation to make sure that storage lends can't be merged into shard 1/2 directly.

Reply:
No, the replicator client is not doing message validations right now. (There's 1000s of messages failing validations right now for various reasons) Right now, we only do hash and signature check